services/src/main/java/org/keycloak/broker/oidc/mappers/AbstractClaimMapper.java

```diff
@@ -122,6 +120,12 @@ public boolean valueEquals(String desiredValue, Object value) {
             for (Object val : list) {
             	if (valueEquals(desiredValue, val)) return true;
             }
+        } else if (value instanceof JsonNode) {
+            try {
+                JsonSerialization.readValue(desiredValue, JsonNode.class).equals(value);
```

Comment:
Isn't there something missing here? Like: "if (JsonSerialization.readValue(desiredValue, JsonNode.class).equals(value)) return true" like in other types? As it is currently, it will always fall to the bottom in case that "value" is JsonNode and method will return false always.

Reply:
Yes, it is missing. PR updated.